### PR TITLE
refactor goodie config option as enum

### PR DESCRIPTION
**Pull request by xuwhite**

### CODE REVIEW

File: config/config.default.php [hunk 1]
```diff
@@ -275,14 +277,11 @@
...
-    // When true changes everything from shirts to goodies and disables shirt size
-    'other_goodie'            => (bool) env('OTHER_GOODIE', false),
-
-    // Enables the goody configuration on signup and profile
-    'enable_goody'            => (bool) env('ENABLE_GOODY', false),
+    // Resembles the Goodie Type. There are three options:
+    // 'none' => no goodie at all
+    // 'goodie' => a goodie which has no sizing options
+    // 'tshirt' => goodie that is called tshirt and has sizing options
+    'goodie_type' => GoodieType::from(env('GOODIE_TYPE', 'goodie')),
```
Comment (MyIgel) [inline config/config.default.php:284]: The only allowed function in config files is env() as it just reads values and static castst. Everything else will confuse users.

The `GoodieType::from` should be used in the code itself instead.

Comment (xuwhite) [reply config/config.default.php:284]: where else?

Comment (xuwhite) [reply config/config.default.php]: done

File: config/routes.php [hunk 2]
```diff
@@ -174,7 +174,7 @@ function (RouteCollector $route): void {
             function (RouteCollector $route): void {
                 // Shirts
                 $route->addGroup(
-                    '/shirt',
+                    '/goodie',
```
Comment (xuwhite) [inline config/routes.php:177]: und für die kleinen changes ein ganzer commit ???
grummel grummel

**Comment by xuwhite** [comment]
git stash hilft gummmel grummel

File: config/routes.php (hunk 2 above)
Comment (MyIgel) [reply config/routes.php:177]: Just to not confuse you...

**Pull request merged**